lib/ai_bot/personas/persona.rb

```diff
@@ -171,6 +176,68 @@ def find_tool(parsed_function)
             persona_options: options[tool_klass].to_h,
           )
         end
+
+        def rag_fragments_prompt(conversation_context)
+          upload_refs =
+            UploadReference.where(target_id: id, target_type: "AiPersona").pluck(:upload_id)
```

Comment:
`id` is null for me here, is this supposed to be `@persona.id` here and elsewhere in this method?

The `id` method you defined on line 87 only works for `system_personas`.